### PR TITLE
Reorg

### PR DESCRIPTION
**Pull request by SatjaSivcev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,3 +32,6 @@ If you need assistance with using Zivid cameras, visit our Knowledge Base at [ht
 
 ## Licence
 Zivid Samples are distributed under the [BSD license](https://github.com/zivid/matlab-samples/blob/master/LICENSE).
+
+[Downsample-url]: https://github.com/zivid/matlab-samples/tree/master/source/Applications/Advanced/Downsample/Downsample.cpp
+[ReadIterateZDF-url]: https://github.com/zivid/matlab-samples/tree/master/source/Applications/Basic/FileFormats/ReadIterateZDF/ReadIterateZDF.cpp
```
Comment (torbsorb) [inline README.md]: .cpp -> .m I guess?

File: source/Applications/Advanced/DownsamplePointCloud.m [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    else
+        disp('Input file is not in ZDF format')
+    end
+catch ME
+    throw(ME)
+end
+
+end
+
+function [X_new, Y_new, Z_new, Image_new] = downsample(X, Y, Z, Image, Contrast, dsf)
```
Comment (torbsorb) [inline source/Applications/Advanced/DownsamplePointCloud.m]: hasn't this been extracted into downsample.m?

**Comment by torbsorb** [comment]
I'm a little unsure as to what examples you want to keep. There are two in the readme, which led me to believe that one sample called the other. However, convertzdf .

I also believe it would make sense to keep zdfread as a common resource. Is it difficult to call functions from modules in other directories? I'm not a Matlab expert, it's been a while since I worked with it, and I don't have it installed.

**Comment by torbsorb** [comment]
I believe it would make sense to place zdfread.m in Applications/Basic/FileFormats and then use addpath in all examples that use it (outside of this folder)
https://se.mathworks.com/matlabcentral/answers/376645-how-to-use-a-function-that-is-not-in-the-same-folder-as-your-current-folder

File: README.md (hunk 1 above)
Comment (SatjaSivcev) [reply README.md]: yes, done

File: source/Applications/Advanced/DownsamplePointCloud.m (hunk 2 above)
Comment (SatjaSivcev) [reply source/Applications/Advanced/DownsamplePointCloud.m]: Opposite. I wanted to minmize the number of samples so I integrated downsample.m into DownsamplePointCloud.m.

I forgot to delete downsample.m - which I did now.

**Comment by SatjaSivcev** [comment]
> 
> 
> Is there a sample missing for convert? Or was the plan to have ReadIterateZDF call it? I believe it would be useful to have a separate sample that can call convertzdf (like you've done for DownsamplePointCloud.m calling downsample.m.

One can just call convertzdf on its own. Also, by calling help convertzdf, one will get the description of how to use the funciton. That is very common to do in Matlab. I don't think a separate script that is using convertzdf function is needed. It was needed for downsample because one has to readzdf downsamplezdf and also visualize the point cloud to understand what is happening.

**Comment by SatjaSivcev** [comment]
> 
> 
> I'm a little unsure as to what examples you want to keep. There are two in the readme, which led me to believe that one sample called the other. However, convertzdf .
> 
> I also believe it would make sense to keep zdfread as a common resource. Is it difficult to call functions from modules in other directories? I'm not a Matlab expert, it's been a while since I worked with it, and I don't have it installed.

I want to keep three samples - two scripts: DownsamplePointCloud and ReadIterateZDF, and one function: convertzdf. Two in Readme was my mistake.

Making a separate file of readzdf and using it as common for multiple scripts/functions -> it would have to be placed in Matlab path somehow. I would have to investigate this a little bit. I was planning to just copy-paste it whenever I make a new sample script that needs it but you are probably right. It would make sense to have it as a separate file used by others.

**Comment by SatjaSivcev** [comment]
> 
> 
> I believe it would make sense to place zdfread.m in Applications/Basic/FileFormats and then use addpath in all examples that use it (outside of this folder)
> https://se.mathworks.com/matlabcentral/answers/376645-how-to-use-a-function-that-is-not-in-the-same-folder-as-your-current-folder

Done for DownsamplePointCloud.m

**Comment by SatjaSivcev** [comment]
Ready for the next round; I will squash/rebase later.

**Comment by SatjaSivcev** [comment]
> 
> 
> I assume you've tested that it runs without errors and thus approve

Yes I have. Cool, mering so.

**Comment by SatjaSivcev** [comment]
Ready for the final review.

This description is not so good. It seems that two samples do the same thing, which is not true. Any ideas how to improve?

image

File: README.md [hunk 3]
```diff
@@ -4,16 +4,18 @@ This repository contains **MATLAB** code samples for **Zivid**.
...
-- [**downsample**](https://github.com/zivid/matlab-samples/blob/master/downsample.m) - Function for downsampling a Zivid point cloud.
+There are two main categories of samples: **Camera** and **Applications**. The samples in the **Camera** category focus only on how to use the camera. The samples in the **Applications** category use the output generated by the camera, such as the 3D point cloud, a 2D image or other data from the camera. These samples shows how the data from the camera can be used.
 
-Note: The functions are independent of any MATLAB toolboxes, while sample scripts require [**Image Processing Toolbox**](https://se.mathworks.com/products/image.html) and [**Computer Vision Toolbox**](https://se.mathworks.com/products/computer-vision.html).
+- **Applications**
+  - **Basic**
+    - **FileFormats**
+      - [**zdfread**][zdfread-url] - This examples shows how to read a Zivid point cloud from a .zdf file.
+      - [**ReadIterateZDF**][ReadIterateZDF-url] - This example shows how to import and display a Zivid point cloud from a .zdf file.
```
Comment (torbsorb) [inline README.md]: Maybe change description to:
"This example shows how to import and display a Zivid point cloud. In addition it shows how to iterate through, and extract individual points. Note that this example depends on [**zdfread**][zdfread-url] to read .zdf files."

Comment (SatjaSivcev) [reply README.md]: I like the first part. I will for now skip 'Note that this example depends on [zdfread][zdfread-url] to read .zdf files.' If we were to put that, we would then need to put the same sentence for other samples that depend on zdfread (Downsample and convertzdf), and that would be verbose.

Comment (SatjaSivcev) [reply README.md]: Done

**Pull request merged**